Health Tracking System

Thiru5 opened the pull request:
- added list of heart images
- added tracking of health points by list of heart images
- added healthcontroller.cs which increments and decrements the list of heart images
- adjusted minor bug in health.cs for respawn health

Thiru5 committed:
Health Tracking System

- added list of heart images
- added tracking of health points by list of heart images
- added healthcontroller.cs which increments and decrements the list of heart images
- adjusted minor bug in health.cs for respawn health

seaerchin commented on Assets/Scripts/Mechanics/HealthController.cs:
```diff
@@ -0,0 +1,39 @@
+
+
+    public void Update(){
+        if(Input.GetKeyDown(KeyCode.Return)){
```
this bugs on keydown of enter etc - so when a player is walking through the map, if this happens, the health will decrement

Thiru5 replied:
removed it in the next update

seaerchin commented on Assets/Scripts/Mechanics/Health.cs:
```diff
@@ -5,11 +5,15 @@
     public class Health : MonoBehaviour
     {
+
+        public PlayerController player;
```
i don't think this circular reference is needed - player can already obtain a ref through using ```GetComponent```

Thiru5 replied:
ok rectified

seaerchin commented on Assets/Scripts/Gameplay/PlayerEnemyCollision.cs:
```diff
@@ -15,6 +15,7 @@ public class PlayerEnemyCollision : Simulation.Event<PlayerEnemyCollision>
     {
         public EnemyController enemy;
         public PlayerController player;
+        public HealthController healthCon;
```
no point having this here cos it's done below and never persisted

Thiru5 replied:
ok rectifed

seaerchin commented on Assets/Scripts/Mechanics/ScoreController.cs:
```diff
@@ -0,0 +1,35 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+using Platformer.Mechanics;
+
+public class ScoreController : MonoBehaviour
+{
+    [SerializeField]
+    private TokenController token;
+    public TMP_Text token_score;
+    public int _enemy_score = 0;
+
+    public void incrementEnemyScore() { 
+        _enemy_score++;
+        Debug.Log("increased" + _enemy_score);
+    }
+
+    public TMP_Text enemy_score;
+    
+    // Start is called before the first frame update
+    void Start()
+    {
+        token_score.text = "0";
+    } 
+
+    // Update is called once per frame
+    // just poll and update
+    void Update()
+    {
+        token_score.text = token.getCount.ToString();
+        enemy_score.text = _enemy_score.ToString();
+        Debug.Log("Enemy" + _enemy_score);
+    }
+}
```
isn't this already in dev?

Thiru5 replied:
yea im not sure why it was tagged as an update LOL

Thiru5 committed:
Merge branch 'development' into issue-4-track

Thiru5 committed:
Update HealthController.cs

- removed update() function

Thiru5 committed:
Rectified minor reference errors

- removed additional ref in health.cs
- removed additional ref in PlayerEnemyCollision
- removed scorecontroller

Thiru5 committed:
Updated Health Tracking

- added Immediate Death function for deathzone

- added immediate death function call to deathzone.cs

Thiru5 committed:
Update .gitignore
